CMakePresets.json

```diff
@@ -0,0 +1,134 @@
+{
+  "version": 3,
+  "cmakeMinimumRequired": {
+    "major": 3,
+    "minor": 20,
+    "patch": 0
+  },
+  "configurePresets": [
+    {
+      "name": "community",
+      "displayName": "Build ArangoDB Community Edition",
+      "binaryDir": "${sourceDir}/build/${presetName}",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Release",
+        "USE_ENTERPRISE": "Off",
+        "USE_JEMALLOC": "On",
+        "USE_IPO": "On"
+      }
+    },
+    {
+      "name": "enterprise",
+      "inherits": "community",
+      "displayName": "Build ArangoDB Enterprise Edition",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Release",
+        "USE_ENTERPRISE": "On",
+        "USE_JEMALLOC": "On",
+        "USE_IPO": "On"
+      }
+    },
+    {
+      "name": "community-developer",
+      "inherits": "community",
+      "displayName": "Build ArangoDB Community Edition (Developer Build)",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "RelWithDebInfo",
+        "USE_IPO": "Off",
+        "USE_MAINTAINER_MODE": "On",
+        "USE_FAIL_ON_WARNINGS": "On",
+	"USE_GOOGLE_TESTS": "On",
+	"USE_FAILURE_TESTS": "On",
+	"USE_STRICT_OPENSSL_VERSION": "Off",
+        "USE_SEPARATE_REPLICATION2_TESTS_BINARY": "On"
```

Comment:
I think there is also USE_SEPARATE_GRAPH_TESTS_BINARY.
my personal preference is to have the separate Test file (or properply split all.)
I would like to have the default of both to be ON or both to be OFF. (My preference is ON)

Reply:
I think I will make the Replication2 and Graph test binaries proper `add_executable`s in their subdirectories and just exclude them from all.

Then they're always configured (negligible cost), and you can always build them if you're developing on that bit;

how does that sound?